src/main/java/backend/resource/TurboLabel.java

```diff
@@ -197,18 +286,65 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         TurboLabel that = (TurboLabel) o;
-        return actualName.equals(that.actualName) && colour.equals(that.colour);
+        return fullName.equals(that.fullName) && colour.equals(that.colour);
     }
 
     @Override
     public int hashCode() {
-        int result = actualName.hashCode();
+        int result = fullName.hashCode();
         result = 31 * result + colour.hashCode();
         return result;
     }
 
     @Override
     public int compareTo(TurboLabel o) {
-        return actualName.compareTo(o.getActualName());
+        return fullName.compareTo(o.getFullName());
+    }
+
+    private static String buildFullName(String group, String name, boolean isExclusive) {
+        return group + (isExclusive ? EXCLUSIVE_DELIMITER : NONEXCLUSIVE_DELIMITER) + name;
+    }
+
+    private Grouping determineGrouping() {
+        if (!getDelimiter(fullName).isPresent()) return Grouping.NONE;
```

Comment:
Do you mean `determineGrouping` ? It is used once during initialization of the grouping